lib/Find-VS2017.cs

```diff
@@ -228,9 +228,10 @@ private static bool CheckInstance(ISetupInstance2 setupInstance2, ref StringBuil
                     hasMSBuild = true;
                 else if (id == "Microsoft.VisualStudio.Component.VC.Tools.x86.x64")
                     hasVCTools = true;
-                else if (id.StartsWith(Win10SDKPrefix))
-                    Win10SDKVer = Math.Max(Win10SDKVer, UInt32.Parse(id.Substring(Win10SDKPrefix.Length)));
-                else if (id == "Microsoft.VisualStudio.Component.Windows81SDK")
+                else if (id.StartsWith(Win10SDKPrefix)) {
+                    string[] parts = id.Substring(Win10SDKPrefix.Length).Split('.');
+                    Win10SDKVer = Math.Max(Win10SDKVer, UInt32.Parse(parts[0]));
```

Comment:
Only the Desktop C++ SDK is good, according to https://docs.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-community it has `.Desktop` appended. What about:

```patch
-                    Win10SDKVer = Math.Max(Win10SDKVer, UInt32.Parse(parts[0]));
+                    if (parts.Length == 1 || (parts.Length == 2 && parts[1] == "Desktop"))
+                        Win10SDKVer = Math.Max(Win10SDKVer, UInt32.Parse(parts[0]));
+                    else
+                        continue;
```